### PR TITLE
[C#] feat: Implemented basic changes needed to support streaming tool calls

### PR DESCRIPTION
**Pull request by Stevenic**
## Linked issues

closes: #2098

## Details

Port the streaming support for tools augmentation to C#.

#### Change details

- Added StreamingChatToolCallsBuilder and SequenceBuilder helper classes.
- Updated OpenAIModel class to process tool call chunks and return them as a message with ActionCalls.
- Updated the LLMClient class to pass through the ActionCalls without closing the streaming response session.

**code snippets**:

**screenshots**:

## Attestation Checklist

- [ ] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

### CODE REVIEW

**Review by lilyydu — COMMENTED** [review]
Code LGTM, could we also push the updates to the lightbot sample?

**Comment by Stevenic** [comment]
I didn't modify the sample but I will

**Comment by singhk97** [comment]
@Stevenic Could you point this to `dotnet-dev` instead. That's the dev branch to avoid have samples in main that aren't compatible with the deployed library bits.

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Clients/LLMClient.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿using Microsoft.Bot.Builder;
+﻿using Google.Protobuf.WellKnownTypes;
```
Comment (singhk97) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Clients/LLMClient.cs:1]: is this being used? if not can you remove it

**Review by singhk97 — APPROVED** [review]

**Pull request merged**